### PR TITLE
Fix explorer client peer dependency warnings

### PR DESCRIPTION
**Pull request by rupurt**
Before:
Screen Shot 2019-08-07 at 5 39 28 PM

After:
Screen Shot 2019-08-07 at 5 38 22 PM

### CODE REVIEW

File: explorer/client/package.json [hunk 1]
```diff
@@ -24,9 +31,9 @@
   },
   "main": "app.js",
   "proxy": "http://localhost:8080",
-  "dependencies": {},
-  "devDependencies": {
-    "@babel/preset-typescript": "^7.3.3",
+  "dependencies": {
+    "@babel/core": "7.4.3",
+    "@babel/preset-typescript": "~7.3.3",
```
Comment (typescribe) [inline explorer/client/package.json]: Why are we restricting this to minor only?

File: explorer/client/package.json [hunk 2]
```diff
@@ -24,9 +31,9 @@
   },
   "main": "app.js",
   "proxy": "http://localhost:8080",
-  "dependencies": {},
-  "devDependencies": {
-    "@babel/preset-typescript": "^7.3.3",
+  "dependencies": {
+    "@babel/core": "7.4.3",
```
Comment (typescribe) [inline explorer/client/package.json]: Any reason why we're pinning this dep but not others?

File: explorer/client/package.json [hunk 3]
```diff
@@ -51,32 +58,47 @@
...
     "react": "^16.8.4",
     "react-dom": "^16.8.4",
     "react-hot-loader": "^4.8.0",
     "react-redux": "^6.0.1",
     "react-resize-detector": "^4.0.5",
-    "react-scripts": "2.1.8",
     "react-snapshot": "^1.3.0",
     "react-time-ago": "^5.0.2",
+    "redux": "^4.0.0",
     "redux-devtools-extension": "^2.13.8",
```
Comment (typescribe) [inline explorer/client/package.json]: This should be in devDeps too if you're moving all of them

File: explorer/client/package.json [hunk 4]
```diff
@@ -51,32 +58,47 @@
...
     "redux-logger": "^3.0.6",
```
Comment (typescribe) [inline explorer/client/package.json]: This should be in devDeps too if you're moving all of them

File: explorer/client/package.json [hunk 5]
```diff
@@ -51,32 +58,47 @@
...
     "redux-object": "^0.5.9",
     "redux-thunk": "^2.3.0",
-    "typescript": "^3.5.0"
-  }
+    "typescript": "^3.5.0",
```
Comment (typescribe) [inline explorer/client/package.json]: This should be in devDeps too if you're moving all of them

File: explorer/client/package.json [hunk 6]
```diff
@@ -51,32 +58,47 @@
...
-    "eslint-plugin-import": "^2.16.0",
-    "eslint-plugin-node": "^8.0.1",
-    "eslint-plugin-prettier": "^3.0.1",
-    "eslint-plugin-promise": "^4.0.1",
-    "eslint-plugin-react": "^7.12.4",
-    "eslint-plugin-standard": "^4.0.0",
     "javascript-time-ago": "^2.0.1",
     "jsbi": "^3.0.0",
     "json-api-normalizer": "^0.4.14",
+    "html-webpack-plugin": "4.0.0-beta.5",
```
Comment (typescribe) [inline explorer/client/package.json]: This should be in devDeps too if you're moving all of them

File: explorer/client/package.json [hunk 7]
```diff
@@ -51,32 +58,47 @@
...
     "module-to-cdn": "^3.1.2",
```
Comment (typescribe) [inline explorer/client/package.json]: This should be in devDeps too if you're moving all of them

File: explorer/client/package.json [hunk 8]
```diff
@@ -51,32 +58,47 @@
     "dynamic-cdn-webpack-plugin": "^4.0.0",
```
Comment (typescribe) [inline explorer/client/package.json]: This should be in devDeps too if you're moving all of them

File: explorer/client/package.json [hunk 9]
```diff
@@ -24,9 +31,9 @@
   },
   "main": "app.js",
   "proxy": "http://localhost:8080",
-  "dependencies": {},
-  "devDependencies": {
-    "@babel/preset-typescript": "^7.3.3",
+  "dependencies": {
+    "@babel/core": "7.4.3",
+    "@babel/preset-typescript": "~7.3.3",
     "@craco/craco": "^5.2.0",
```
Comment (typescribe) [inline explorer/client/package.json]: This should be in devDeps too if you're moving all of them

File: explorer/client/package.json (hunk 2 above)
Comment (typescribe) [reply explorer/client/package.json]: Also, this should be in `devDeps`

File: explorer/client/package.json [hunk 10]
```diff
@@ -51,32 +58,47 @@
...
+    "eslint-plugin-react": "^7.12.4",
+    "eslint-plugin-standard": "^4.0.0",
+    "react-scripts": "3.0.1",
+    "prettier": "^1.16.4",
+    "prettier-config-standard": ">= 1.0.0"
+  },
+  "resolutions": {
+    "caniuse-lite": "1.0.30000957"
+  },
+  "comments": {
```
Comment (typescribe) [inline explorer/client/package.json]: Don't think this is a valid package.json key. We should move this out to the README or something

File: explorer/client/src/containers/JobRuns/Index.tsx [hunk 11]
```diff
@@ -81,21 +96,23 @@ const jobRunsSelector = ({
...
+  return {
+    rowsPerPage: 10,
+    query: state.search.query,
+    jobRuns: jobRunsSelector(state),
+    count: jobRunsCountSelector(state)
+  }
+}
 
-const mapDispatchToProps = (dispatch: Dispatch<any>) =>
+const mapDispatchToProps = (dispatch: Dispatch): DispatchProps =>
```
Comment (typescribe) [inline explorer/client/src/containers/JobRuns/Index.tsx]: There should be a `MapDispatchToProps` exported from `react-redux` that you can annotate this function with.

File: explorer/client/src/containers/JobRuns/Index.tsx [hunk 12]
```diff
@@ -81,21 +96,23 @@ const jobRunsSelector = ({
...
   return state.jobRunsIndex.count
 }
 
-const mapStateToProps = (state: IState) => ({
-  rowsPerPage: 10,
-  query: state.search.query,
-  jobRuns: jobRunsSelector(state),
-  count: jobRunsCountSelector(state)
-})
+const mapStateToProps = (state: State): StateProps => {
```
Comment (typescribe) [inline explorer/client/src/containers/JobRuns/Index.tsx]: There should be a `MapStateToProps` exported from `react-redux` that you can annotate this function with.

File: explorer/client/src/containers/JobRuns/Show.tsx [hunk 13]
```diff
@@ -111,25 +123,21 @@ const jobRunSelector = (
...
+const etherscanHostSelector = ({ config }: State) => {
   return config.etherscanHost
 }
 
-interface IOwnProps {
-  jobRunId?: string
-}
-
-const mapStateToProps = (state: IState, { jobRunId }: IOwnProps) => {
+const mapStateToProps = (state: State, { jobRunId }: OwnProps) => {
```
Comment (typescribe) [inline explorer/client/src/containers/JobRuns/Show.tsx]: Annotate this with the `MapStateToProps` type from `react-redux`

File: explorer/client/src/containers/JobRuns/Show.tsx [hunk 14]
```diff
@@ -111,25 +123,21 @@ const jobRunSelector = (
...
   const jobRun = jobRunSelector(state, jobRunId)
   const etherscanHost = etherscanHostSelector(state)
 
   return { jobRun, etherscanHost }
 }
 
-const mapDispatchToProps = (dispatch: Dispatch<any>) =>
+const mapDispatchToProps = (dispatch: Dispatch) =>
```
Comment (typescribe) [inline explorer/client/src/containers/JobRuns/Show.tsx]: Annotate this with the `MapDispatchToProps` type from `react-redux`

File: explorer/client/package.json (hunk 2 above)
Comment (rupurt) [reply explorer/client/package.json]: It's not ejected so fails to boot when not in sync with https://github.com/facebook/create-react-app/blob/v3.0.1/packages/react-scripts/package.json#L31

File: explorer/client/package.json (hunk 10 above)
Comment (rupurt) [reply explorer/client/package.json]: From what I could tell on the intertubes you can use any key as long as it's not reserved https://stackoverflow.com/questions/10065564/add-custom-metadata-or-config-to-package-json-is-it-valid. 

I'm not super tied to having the comment here but I do like that it's located close to it.

File: explorer/client/package.json (hunk 1 above)
Comment (rupurt) [reply explorer/client/package.json]: It was conflicting with the definition in the root `package.json` https://github.com/smartcontractkit/chainlink/blob/develop/package.json#L30

File: explorer/client/package.json (hunk 10 above)
Comment (typescribe) [reply explorer/client/package.json]: sgtm, lets keep it there then

File: explorer/client/package.json (hunk 2 above)
Comment (typescribe) [reply explorer/client/package.json]: hm, maybe we should leave it unpinned, and instead add a yarn `resolution` key

File: explorer/client/package.json (hunk 1 above)
Comment (typescribe) [reply explorer/client/package.json]: Lets use yarn `resolution` for this?

File: explorer/client/package.json (hunk 2 above)
Comment (rupurt) [reply explorer/client/package.json]: Can you expand on why resolution should be used here?

**Pull request merged**